File: limetech_components.client/src/app/models/component.dto.ts
import { PartStatus } from "./status.dto";

export interface ComponentDTO {

  id: number;
  name: string | null;
  typeOfProduct: string | null;
  imageUrl: string | null;
  price: number;
  purchasedCount: number;
  productionYear: number;
  powerUsage: number;
  status: PartStatus;
  stockCount: number;
  isPublic: boolean;

}

File: LimeTech_Components.Server/DTOs/ComponentDTO.cs
using LimeTech_Components.Server.Data.Models;

namespace LimeTech_Components.Server.DTOs
{
    public class ComponentDTO
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Producer { get; set; } = null!;
        public required string TypeOfProduct { get; set; }
        public string ImageUrl { get; set; } = null!;
        public decimal Price { get; set; }
        public int ProductionYear { get; set; }
        public int PowerUsage { get; set; }
        public PartStatus Status { get; set; }
    }

}
